### ISSUE TITLE
CLI/Tooling update 

### ISSUE THREAD

**Issue by cueckoo** [cue-lang/cue#337]
_Originally opened by @myitcv in https://github.com/cuelang/cue/issues/337_

The `cue` command just received a major update with [release v0.1.0](https://github.com/cuelang/cue/releases/tag/v0.1.0). There are still various thing that this release prepare for, but were left unaddressed:

In the shorter term we intent to target the following:
* new "default" command, invoked by just `cue`, replacing `eval` and making `export` and `def` special cases.
* `cue test` to test CUE packages and `cue` tool commands.
* `cue diff/snapshot` for comparing between two CUE versions.
* `cue run` a more sensible rewrite of `cue cmd`.
* The latter may include what constitutes a tool file and local import of packages.
* Support for dowload of OpenAPI and JSON Schema references in `cue import/get`
* Allowing builtin tooling layer functions (not just task types) to allow implementing a `filepath` package as well as other functionality.

More medium term targets:
* MVS implementation for CUE modules, allowing versioning and importing of CUE packages analogous to Go.
* Task type to allow writing Go functions in the CUE tooling layer, automatically converting CUE types to equivalent Go.
* Generic task adaptor support.
* More dynamic task evaluation, allowing dynamically created tasks.

Optional:
* User-defined flags. This may not be needed with the addition of `--inject` support.
* "Merging" of packages on the command line, for instance using the `-m` flag. It is not entirely clear what the semantics should be there and if it is a good idea at all.

### Links

- [All issues specific to this roadmap issue](https://github.com/cue-lang/cue/issues?q=label%3Aroadmap%2Fcli+)
- [Open issues specific to this roadmap issue](https://github.com/cue-lang/cue/labels/roadmap%2Fcli)

**Comment by dolanor** [comment]
Links at the bottom of the OP are broken

**Comment by slewiskelly** [comment]
Yes, the issue was imported from the original repo. Assume these to be the equivalent:
- [All](https://github.com/cue-lang/cue/issues?q=label%3Aroadmap%2Fcli+)
- [Open](https://github.com/cue-lang/cue/labels/roadmap%2Fcli)

**Comment by myitcv** [comment]
Thanks, @slewiskelly - updated with your links.
